Sources/Fuzzilli/Compiler/Compiler.swift

```diff
@@ -486,6 +485,45 @@ public class JavaScriptCompiler {
                 try compileBody(withStatement.body)
             }
             emit(EndWith())
+        case .switchStatement(let switchStatement):
+            // Precompute tests because between BeginSwitch and BeginSwitchCase we can't emit instructions
```

Comment:
I think it can lead to different behavior, for example:
```js
function foo(i) {
    print(i);
    return i;
}

switch (X) {
    case foo(0): break;
    case foo(1): break;
    case foo(2): break;
    default: break;
}
```
For `X == 1`, this will print 0 and 1. But after compilation to FuzzIL, the sample will now be something like
```js
function foo(i) {
    print(i);
    return i;
}

let v0 = foo(0);
let v1 = foo(1);
let v2 = foo(2);

switch (X) {
    case v0: break;
    case v1: break;
    case v2: break;
    default: break;
}
```
And so always print 0, 1, 2. So there's an observable difference.

Reply:
Now documented as #441